### PR TITLE
TS protocol

### PR DESCRIPTION
**Pull request by buoyad**
r? @keybase/react-hackers

### CODE REVIEW

File: shared/constants/types/rpc-gen.tsx [hunk 1]
```diff
@@ -0,0 +1,4107 @@
...
+  | 0 // MAIN_0
+  | 1 // CHAT_1
+
+export type DbValue = Bytes
+export type DeletedTeamInfo = {readonly teamName: String; readonly deletedBy: String; readonly id: Gregor1.MsgID}
+export type DesktopStatus = {readonly version: String; readonly running: Boolean; readonly log: String}
+export type Device = {readonly type: String; readonly name: String; readonly deviceID: DeviceID; readonly cTime: Time; readonly mTime: Time; readonly lastUsedTime: Time; readonly encryptKey: KID; readonly verifyKey: KID; readonly status: Int}
+export type DeviceDetail = {readonly device: Device; readonly eldest: Boolean; readonly provisioner?: Device | null; readonly provisionedAt?: Time | null; readonly revokedAt?: Time | null; readonly revokedBy: KID; readonly revokedByDevice?: Device | null; readonly currentDevice: Boolean}
+export type DeviceEk = {readonly seed: Bytes32; readonly metadata: DeviceEkMetadata}
+export type DeviceEkMetadata = {readon
```
Comment (chrisnojima) [inline shared/constants/types/rpc-gen.tsx]: spoke IRL but i think the enum and this type of type can be merged into one thing now

**Comment by buoyad** [comment]
@chrisnojima thanks, done!

File: protocol/bin/flow.js [hunk 2]
```diff
@@ -81,7 +81,7 @@ Promise.all(
       }, {})
       .then(typeDefs => {
         writeFlow(typeDefs, project)
-        write(typeDefs, project)
+        // write(typeDefs, project)
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: 🔥

File: protocol/bin/flow.js [hunk 3]
```diff
@@ -153,7 +153,8 @@ function analyzeTypes(json, project) {
         map[t.name] = `export type ${t.name} = ${parseRecord(t)}`
         break
       case 'enum':
-        map[t.name] = `export type ${t.name} =${parseEnum(t)}`
+        // no separate enum type needed in ts
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: 🔥

**Pull request merged**